app/assets/javascripts/discourse/app/components/glimmer-site-header.gjs

```diff
@@ -0,0 +1,518 @@
+      if (this.site.mobileView) {
+        this._swipeEvents.addTouchListeners();
+        this._headerWrap.addEventListener("swipestart", this.onSwipeStart);
+        this._headerWrap.addEventListener("swipeend", this.onSwipeEnd);
+        this._headerWrap.addEventListener("swipecancel", this.onSwipeCancel);
+        this._headerWrap.addEventListener("swipe", this.onSwipe);
+      }
+    }
+  }
+
+  _handleArrowKeysNav(event) {
+    const activeTab = document.querySelector(
+      ".menu-tabs-container .btn.active"
+    );
+    if (activeTab) {
+      let activeTabNumber = Number(
+        document.activeElement.dataset.tabNumber || activeTab.dataset.tabNumber
+      );
+      const maxTabNumber =
+        document.querySelectorAll(".menu-tabs-container .btn").length - 1;
+      const isNext = event.key === "ArrowDown";
+      let nextTab = isNext ? activeTabNumber + 1 : activeTabNumber - 1;
+      if (isNext && nextTab > maxTabNumber) {
+        nextTab = 0;
+      }
+      if (!isNext && nextTab < 0) {
+        nextTab = maxTabNumber;
+      }
+      event.preventDefault();
+      document
+        .querySelector(
+          `.menu-tabs-container .btn[data-tab-number='${nextTab}']`
+        )
+        .focus();
+    }
+  }
+
+  @action
+  animateMenu() {
+    const menuPanels = document.querySelectorAll(".menu-panel");
+
+    if (menuPanels.length === 0) {
+      this._animate = this.site.mobileView || this.site.narrowDesktopView;
+      return;
+    }
+
+    let viewMode =
+      this.site.mobileView || this.site.narrowDesktopView
+        ? "slide-in"
+        : "drop-down";
+
+    menuPanels.forEach((panel) => {
+      if (menuPanelContainsClass(panel)) {
+        viewMode = "drop-down";
+        this._animate = false;
+      }
+
+      const cloakElement = document.querySelector(".header-cloak");
+
+      panel.classList.remove("drop-down");
+      panel.classList.remove("slide-in");
+      panel.classList.add(viewMode);
+
+      if (this._animate) {
+        let animationFinished = null;
+        let finalPosition = PANEL_WIDTH;
+        this._swipeMenuOrigin = "right";
+        if (
+          (this.site.mobileView || this.site.narrowDesktopView) &&
+          panel.parentElement.classList.contains(this.leftMenuClass)
+        ) {
+          this._swipeMenuOrigin = "left";
+          finalPosition = -PANEL_WIDTH;
+        }
+        animationFinished = panel.animate(
+          [{ transform: `translate3d(${finalPosition}px, 0, 0)` }],
+          {
+            fill: "forwards",
+          }
+        ).finished;
+
+        if (isTesting()) {
+          waitForPromise(animationFinished);
+        }
+
+        cloakElement.animate([{ opacity: 0 }], { fill: "forwards" });
+        cloakElement.style.display = "block";
+
+        animationFinished.then(() => {
+          if (isTesting()) {
+            this._animateOpening(panel);
+          } else {
+            discourseLater(() => this._animateOpening(panel));
+          }
```

Comment:
as before: should later be replaced with next, and do we need the separate path for testing?

Reply:
this was copied over, I would prefer to address it in a separate PR